proxy/http2/Http2ClientSession.cc

```diff
@@ -299,18 +299,38 @@ Http2ClientSession::set_half_close_local_flag(bool flag)
 }
 
 int64_t
-Http2ClientSession::xmit(const Http2TxFrame &frame)
+Http2ClientSession::xmit(const Http2TxFrame &frame, bool flush)
 {
   int64_t len = frame.write_to(this->write_buffer);
+  this->_pending_sending_data_size += len;
+
+  // Force flush for some cases
+  if (!flush) {
+    // Flush if we already use half of the buffer to avoid adding a new block to the chain.
```

Comment:
Did you check this behavior? I bit doubt calling `flush()` prevents adding a new block. Because `write_reenable()` doesn't do actual write.

Reply:
You're right. There's no guarantee, and "flush" may be not a good name in that sense. Like the big frame case explained on next line, `write_buffer` may have multiple blocks, but it's just an unfortunate case (it always has been), so it doesn't affect the sending process.

Reply:
But the write_reenable gets the writing underway, correct?  So it does ensure that we don't buffer up all the writes until the End of Stream.

Reply:
Correct, and there is also receive window on H2 layer. Basically, the total size of the buffered data will not be bigger than window size. Strictly speaking, it can be bigger if we send H2 frames that are not under flow control, but the sizes are vary small compare to DATA frame.

Reply:
This is a wild guess, but we can achieve the same performance by just bumping the buffer size? (without these pending data tracking and flush functions)

Reply:
That might be better than current, but I think it would call SSLWriteBuffer more frequently than the one with pending mechanism, because the buffered data would be sent regardless of the amount of the data. The pending mechanism ensure such small writes don't happen.